nixos/modules/services/networking/iwd.nix

```diff
@@ -4,16 +4,142 @@ with lib;
+    echo "=$(printf "$1" | ${pkgs.unixtools.xxd}/bin/xxd -pu)"
+  '';
+
+  # If a network consists of alphanumeric chars and `_` and `-`, the network will be stored
+  # as `/var/lib/iwd/<ssid>.<ext>`. In any other case a hex-encoded version of the SSID will
+  # be stored with an `=` as prefix.
+  encodeIfNeeded = file: ext: let
+    allowedChars = stringToCharacters "01234556789abcdefghijklmnopqrstuvwxyz-_ ";
+    skipEncoding = all (char: elem (toLower char) allowedChars) (stringToCharacters file);
+  in
+    "/var/lib/iwd/${if skipEncoding then file else "$(${encodeFileName} \"${file}\")"}.${ext}";
+
+  # Basically `types.str`, but the `check` function validates the string-length as well.
+  fixedStringLength = min: max: mkOptionType {
+    name = "str-with-length";
+    description = if min == max
+      then "string with exactly ${toString min} chars."
+      else "string which must have at least ${toString min} and at most ${toString max} chars.";
+    inherit (types.str) merge;
+    check = x: isString x && stringLength x >= min && stringLength x <= max;
+  };
+
+  # The file extension of the network-config determines the type of the network. Currently,
+  # we support public networks (`open`) and networks protected with a passphrase (`psk`).
+  # Anything else, like 802.1x networks are not supported by this module.
+  determineType = cfg: if cfg.psk != null || cfg.passphrase != null then "psk" else "open";
+
+  # Generates a name-value-pair with configurations for a SSID. For further information
+  # about the configuration options, please refer to the upstream docs:
+  #
+  # https://iwd.wiki.kernel.org/networkconfigurationsettings
+  predefinedNetworks = flip mapAttrs cfg.networks (name: cfg: {
+    extension = determineType cfg;
+
+    # Cannot use the SSID here: the hex-encode is done at build-time, not eval time
+    # and several special chars in a file name would break `writeText`.
+    file = pkgs.writeText "ssid" (generators.toINI {} ({
+      Settings.AutoConnect = cfg.autoConnect;
+      Settings.Hidden = cfg.hidden;
+    } // (optionalAttrs (cfg.psk != null) {
+      Security.PreSharedKey = cfg.psk;
+    }) // (optionalAttrs (cfg.passphrase != null) {
+      Security.Passprase = cfg.passphrase;
+    })));
+  });
 in {
-  options.networking.wireless.iwd.enable = mkEnableOption "iwd";
+  options.networking.wireless.iwd = {
+    enable = mkEnableOption "iwd";
+
+    interfaces = mkOption {
+      type = types.nullOr (types.listOf types.str);
+      example = [ "wlp2s0" ];
+      apply = x: if x == null then x else concatStringsSep " " x;
+      default = null;
+      description = ''
+        List of interfaces to use.
+      '';
+    };
+
+    networks = mkOption {
+      default = {};
+      example = {
+        "karlsruhe.freifunk.net" = {};
+        "secured".passphrase = "12345678";
+        "network-with-psk".psk = "bf4c086e055e492373a0f59ff614f61fc2d597a56db94e17a165f9ef1d42c066";
+      };
```

Comment:
For examples to render on multiple lines you need to wrap it in a string and use `lib.literalExample`:
```nix
{
  example = lib.literalExample ''
    {
      "karlsruhe.freifunk.net" = {};
      # ...
    }
  '';
}
```

Reply:
Good point! :)